lib/optimizer/level-2/simplify-variables.js

```diff
@@ -0,0 +1,33 @@
+function simplifyVariables(tokens) {
+    var usedVariables = {};
+
+    for (var i = tokens.length - 1; i >= 0; i--) {
+        if (tokens[i][0] === 'rule') {
+            var stringifiedRule = JSON.stringify(tokens[i][1]);
```

Comment:
A simple way to keep track of variables in different rules. Maybe it is a bit dirty?